swarm/runtime/statsdb/ingestion.py

```
@@ -558,162 +638,189 @@ def ingest_events(self, events: List[Dict[str, Any]], run_id: str) -> int:
+            # This handles any edge cases or connection weirdness
+            actual_new_events = []
+            for event in new_events:
+                if self._insert_raw_event(event):
+                    self._project_event(event, run_id)
+                    actual_new_events.append(event)
+            # Update new_events to reflect what was actually inserted in fallback mode
+            new_events = actual_new_events
+
+        return len(new_events)
+
+    def _project_event(self, event: Dict[str, Any], run_id: str):
```

Ruff (0.15.0)

[warning] Missing return type annotation for private function `_project_event`

Add return type annotation: `None`

(ANN202)